### ISSUE TITLE
CMIP MPI-ESM version 20190710 is retracted/no longer available, snakefile script 404's

### ISSUE THREAD

**Issue by mjwillson** [pangeo-data/WeatherBench#39]
It appears the files downloaded by the Snakefile for CMIP no longer exist at http://esgf3.dkrz.de/. Based on information at:
https://esgf-data.dkrz.de/search/cmip6-dkrz/?mip_era=CMIP6&activity_id=CMIP&source_id=MPI-ESM1-2-HR&institution_id=MPI-M&experiment_id=historical&variant_label=r1i1p1f1&table_id=6hrPlevPt

It seems the only version available is v20190815, and the version v20190710 used in the scripts isn't available any more.
If I tick "Show all versions" I see that:

> IMPORTANT: this dataset has been retracted and is no longer available for download
> Data Node: esgf3.dkrz.de
> Version: 20190710

Do you know what the problem was with this older version, and do you plan to update your derived datasets to the newer version?

In particular for the CMIP fields for which you haven't yet released your derived regridded data (https://github.com/pangeo-data/WeatherBench/issues/32#issuecomment-882486544) do you have any plans to release data based on the older version, newer version?

**Comment by mjwillson** [comment]
To clarify, I don't particularly need to replicate the exact numbers in the paper for CMIP, so no worries if you don't have the original v20190710 data still lying around. I'm just asking in case this means we would need to redownload and regrid the newer v20190815 data ourselves to get a full and consistent set of fields to experiment with CMIP pre-training.

**Comment by mjwillson** [comment]
To further complicate things, it seems the new v20190815 version doesn't include precipitation_flux (pr) or toa_incident_solar_radiation (rsdt).

So unless you're able to release your copies of the v20190710 data (regridded or otherwise) it seems there's no way to replicate (even approximately) the CMIP pretraining using these fields?

**Comment by raspstephan** [comment]
Hi, first of all sorry for the late reply. This indeed seems like an issue. I just checked and I do have the data still available, except for solar radiation for some reason. I am not sure why that didn't end up on the data server. However, it's not suuuper easy to get the data on there because I can't do it without permissions. This means it could take a while (unfortunately very busy right now) and I wouldn't have all the variables anyway.

But I did just check and from what I can see pr and rsdt are still available for the newer version. They are just not under 6hrPlevPt but under E3hr. This probably will require some changes in the code. I am really sorry for the inconvenience. But I think your best bet would be to try modifying the script to download the latest version of the CMIP runs. BTW, the most up to date version of the snakemake file is here: https://github.com/raspstephan/WeatherBench/tree/master/snakemake_configs/MPI-ESM

**Comment by mjwillson** [comment]
Hi Stephan, thanks for the response and the pointers.

I asked about the reasons for the retraction and replacement version and got the following response:

> Corrected datasets are always from the
> same model run, with the only difference being that the data has been
> post-processed in a different (now hopefully correct) way. Thus it is ok
> to combine different variables from different versions, as long as for
> each individual variable, the latest available version is used.

They also pointed me to the errata here: https://errata.es-doc.org/static/index.html. It looks like these were the issues that were corrected in the 20190815 version:

For the 2m_temperature (tas) variable:

> [Wrong variable submitted as tas_6hrPlevPt](https://errata.es-doc.org/static/view.html?uid=9fa7bf63-9bf7-5fbe-9930-bf35563a3fc8)
> --
> Instead of near surface temperature, meridional near surface wind has been published as tas_6hrPlevPt due to a mix up of GRIB codes while standardizing the data. A temporary work around was, using tas_3hr as a substitute: "cdo selhour,0,6,12,18 infile outfile".

And for the geopotential (zg), specific_humidity (hus), temperature (ta), u_component_of_wind (ua), v_component_of_wind (va) variables:

> [Mixed up instantaneous and mean version of the variable](https://errata.es-doc.org/static/view.html?uid=0ec4a5be-6a80-9900-af55-f5d1afeceef6)
> --
> For several 3 and 6-hourly atmospheric variables, the instantaneous and mean version of the respective variables have been accidentally swapped. This happened as for the simulations in question both versions of the variables have been written to the same output streams. Affected variables by MIP table: \| E3hrPt: ua, va, wap, ta \| E3hr: uas, vas, psl \| CF3hr: ta, psl \| 6hrLev: ua, va \| 6hrPlev: uas, vas, wap, tas, psl \| 6hrPlevPt: ua, va, ta, psl, hus, zg, zg500 \| Affected are the DECK and historical simulations of the models "MPI-ESM1-2-LR" and "MPI-ESM1-2-HR" as well as the ScenarioMIP and RFMIP simulations of the model "MPI-ESM1-2-LR".

This might mean that the 2m_temperature used in your paper for CMIP was using the wrong data, and the others above (if obtained from 6hrPlevPt) would have been 6 hour temporal means instead of instantaneous values. Could you confirm?

It looks like CMIP pre-training worked pretty well regardless, so hopefully it'll work even better with the updated fields :)

**Comment by raspstephan** [comment]
Thanks so much for looking this up. It seems indeed that the data wasn't quite correct. Very interesting. But it also means that you could download the corrected data using the correct label if you wanted to, right?

**Comment by mjwillson** [comment]
Thanks, yes the corrected data is available, although for some fields there's a discrepancy between what's available for CMIP (MPI-ESM1.2-HR) and the ERA5 field -- just noting here for reference:

* ERA5 has `total_precipitation` which is a 1-hour accumulation in units of m. CMIP has `pr` in tables `6hrPlev` and `3hr`, which is respectively a 6-hour or 3-hour temporal mean, in units of m/s. So it's won't be possible to get 1-hour accumulations for CMIP, but we can compute the 6-hour accumulations that are used as targets.
* ERA5 has `total_cloud_cover`, which is instantaneous. Closest CMIP field appears to be `clt` in the `3hr` table which is a 3-hour temporal mean.
* ERA5 has `toa_incident_solar_radiation`, which is a 1-hour accumulation in J/m^2. Closest CMIP field appears to be `rsdt` in the `E3hr` table, which is a 3-hour temporal mean in units of W/m^2.

For the other fields it looks like the corrected CMIP data matches the ERA5 values which are instantaneous.

And links I found useful:
http://clipc-services.ceda.ac.uk/dreq/index/miptable.html
https://confluence.ecmwf.int/pages/viewpage.action?pageId=82870405#ERA5:datadocumentation-Parameterlistings
